### ISSUE TITLE
Cannot successfully register targets if VPC subnets are different from VPC CIDR block associations

### ISSUE THREAD

**Issue by nullren** [kubernetes-sigs/aws-load-balancer-controller#3714]
**Describe the bug**
the controller cannot register targets in targetgroupbinding with aws api error

```
ValidationError: The Availability Zone is required for IP address '10.4.40.87' because it is not in the VPC
```

this is because the [registerPodEndpoints](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/9e9432b654461a3bbc92638afcd8f4abc5de977f/pkg/targetgroupbinding/resource_manager.go#L386-L422) function is looking up cidr blocks associated to the vpc and not the actual subnets which is what the aws api compares.

**Steps to reproduce**

create vpc and associate more cidr blocks than subnets

```
$ aws ec2 describe-vpcs --vpc-ids vpc-0ea5e9af6c340f01f | jq -r '.Vpcs[] | .CidrBlockAssociationSet[] | .CidrBlock'
10.1.0.0/16
10.2.0.0/16
10.3.0.0/16
10.4.0.0/16
```

then add a few subnets to the vpc

```
$ aws ec2 describe-subnets --filters "Name=vpc-id,Values=vpc-0ea5e9af6c340f01f" | jq -r '.Subnets[] | .CidrBlock'
10.1.0.0/19
10.2.0.0/19
```

try to manually add a target to the target group

```
$ aws elbv2 register-targets --target-group-arn arn:aws:elasticloadbalancing:us-east-1:000000000000:targetgroup/k8s-default-targetgroupname-0000000/0000000000000 --targets Id=10.4.46.99,Port=80

An error occurred (ValidationError) when calling the RegisterTargets operation: The Availability Zone is required for IP address '10.4.46.99' because it is not in the VPC
```

**Expected outcome**
this is fixed when adding `AvailabilityZone=all` to the request and what should happen in the initial check

**Environment**

* AWS Load Balancer controller version: 2.7.2
* Kubernetes version
* Using EKS (yes/no), if so version? yes

**Additional Context**:

**Comment by nullren** [comment]
i've started a [pr](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3722) to draft out some ideas here but to also on my end to unblock my work.

one thing i'm ensure of is the "source of truth" for what the problem is, it seems there are a few things going on here:
- does the `register-targets` api actually look to just the subnets for discovery?
- is the difference between what the `describe-vpcs` subnets and the subnets i can see from `describe-subnets` something that _**should**_ be reconciled via something like sharing subnets between accounts?
- how important is autodiscovery here and are we going to cause problems by setting `AvailabilityZone=all` for these? is there a reason we should not always set it?

**Comment by AbeOwlu** [comment]
@nullren , if I understand correctly - the target `10.4.46.9` is (not) in a sub-network?
Or the subnet the target is in `10.4.46.9` is not part of the AZs currently registered with the ELB -`10.1.0.0/19 10.2.0.0/19`?

That would be an inadvertent secure implementation feature, right? - The ELB would never really be able to route traffic to target `10.4.46.9` Az(?) from say `10.1.0.0/19`Az1 and `10.2.0.0/19` Az2

**Comment by PT-GD** [comment]
I'm running into a more fundamental problem where the reconciler looks up the VPC of the nodes even in IP mode and appears to put the target groups in that VPC, even when that is a different VPC than the NLB. This causes the reconciler to fail because it cannot create the listener for a target group in a different VPC.

I'm connected via TGW from the AWS LB subnet to the node subnet, and when I manually build an NLB, it works fine. Any thoughts?
